### PR TITLE
Update PyNE to use PyMOAB instead of PyTAPS with newest MOAB

### PR DESCRIPTION
**Pull request by gonuke**
This is a large PR developed collectively by @CNERG to remove the dependency on the deprecated PyTAPS project and replace it with the native python support in MOAB, known as pymoab.

This is the culmination of many months of effort, primarily by @pshriwise, the author of pymoab, and completed by @cnerg over the last month.  

All changes have been internally reviewed, discussed and revised within the @cnerg team.

Details can be found in the news file included in this PR.  Most notably,
* some syntax for mesh and tags has been changed, generally to a more streamlined format
* some dependencies have been more narrowly confined
* new dockerfiles have been added to make it easier to update the CI docker images

Full support for Python 3 should now be possible but is not yet included in this PR.

Any partial review is appreciated and the @cnerg team is, of course, happy to respond to any questions.

This will probably demand a rev of the PyNE version string and a release.

### CODE REVIEW

File: CMakeLists.txt
Comment (katyhuff) [inline CMakeLists.txt]:
```suggestion
# Look for MOAB if requested
```

File: docker/make_pyne_docker_image.py
Comment (katyhuff) [inline docker/make_pyne_docker_image.py]:
```suggestion
    DockerHub(Docker login is assumed...)
```

File: docker/make_pyne_docker_image.py
Comment (katyhuff) [inline docker/make_pyne_docker_image.py]: delete unnecessary blank line
```suggestion
```

File: docker/make_pyne_docker_image.py
Comment (katyhuff) [inline docker/make_pyne_docker_image.py]: For consistency
```suggestion
    description = 'Build a docker image for PyNE'
```

File: docker/make_pyne_docker_image.py
Comment (katyhuff) [inline docker/make_pyne_docker_image.py]: consistency
```suggestion
    name = 'Set docker imgae name'
```

File: docker/make_pyne_docker_image.py
Comment (katyhuff) [inline docker/make_pyne_docker_image.py]:
```suggestion
    push = 'Push docker image on dockerhub'
```

File: pyne/dagmc.pyx
Comment (katyhuff) [inline pyne/dagmc.pyx]: I think PEP8 discourages extra blank lines, and PEP257 discourages extra blank lines in the docstrings.

```suggestion
```

File: pyne/mesh.py
Comment (katyhuff) [inline pyne/mesh.py]:
```suggestion
                - From a PyMOAB instance with multiple entity sets by
```

File: pyne/mesh.py
Comment (katyhuff) [inline pyne/mesh.py]: Something weird is happening, grammatically here.... maybe this is right:


```suggestion
            mesh is a vector of floats in the following form:
```

File: pyne/mesh.py
Comment (katyhuff) [inline pyne/mesh.py:755]: This line seems long. (PEP8 check?)

File: pyne/mesh.py
Comment (katyhuff) [inline pyne/mesh.py:449]: This line seems long. (PEP8 check?)

File: pyne/mesh.py
Comment (katyhuff) [inline pyne/mesh.py]: Unnecessary blank line (see pep8 and pep257)
```suggestion
```

PEP257 recommends one blank line after a class docstring. 

> Insert a blank line after all docstrings (one-line or multi-line) that document a class -- generally speaking, the class's methods are separated from each other by a single blank line, and the docstring needs to be offset from the first method by a blank line.

File: pyne/r2s.py
Comment (katyhuff) [inline pyne/r2s.py]: I don't think a blank line is advisable here. For the docstring to be read properly by many autodoc tools, it really needs to be right after the def line.

```suggestion
```

**Review by katyhuff — APPROVED** [review]
Looks good. I have left a few suggested typo fixes, but otherwise this PR looks very clean. Nice work!

**Comment by katyhuff** [comment]
@pep8speaks suggest diff

File: docker/make_pyne_docker_image.py
Comment (gonuke) [reply docker/make_pyne_docker_image.py]:
```suggestion
    name = 'Set docker image name'
```

File: docker/make_pyne_docker_image.py
Comment (makeclean) [inline docker/make_pyne_docker_image.py]: typo , -> /

File: docker/ubuntu_18.04-dev.dockerfile
Comment (makeclean) [inline docker/ubuntu_18.04-dev.dockerfile:47]: does pip always point to the right place? I often need to use pip3?

**Review by makeclean — APPROVED** [review]
Guys, this truly is a huge an momentous piece of work - my sincere apologies that its taken me so long to review! When I make it back to Madison I owe you guys beer :)

File: docker/ubuntu_18.04-dev.dockerfile
Comment (gonuke) [reply docker/ubuntu_18.04-dev.dockerfile:47]: I think the above `update-alternatives` will guarantee that for a py3 build

File: docker/make_pyne_docker_image.py
Comment (gonuke) [reply docker/make_pyne_docker_image.py]: This isn't a typo, but maybe it's not clear, the name will be one of:
* `<basename>_moab`
* `<basename>_dagmc`
* `<basename>_dagmc_pymoab`

Possibly `_pyne_deps` will also be appended.

In retrospect, this is an odd combination of options... perhaps we should review.

Comment (gonuke) [reply docker/make_pyne_docker_image.py]:
```suggestion
    will be "pyne/ubuntu_18.04{_moab/_dagmc{_pymoab}}{_pyne-deps/}"
```

Comment (gonuke) [reply docker/make_pyne_docker_image.py]: It'd be great if this didn't hold up the merge and we addressed this in a follow-up

**Comment by gonuke** [comment]
Thanks for the reviews @katyhuff & @makeclean - If someone's up to merge this, we should discuss pushing a new version.

**Comment by makeclean** [comment]
Lets go for the merge :)

**Pull request merged**
